src/sanitizer.js

```diff
@@ -213,6 +248,98 @@ const INVALID_INLINE_STYLE_REGEX =
  * @return {string}
  */
 export function sanitizeHtml(html) {
+  if (isExperimentOn(self, 'svg-in-mustache')) {
+    return purifyHtml(html);
+  } else {
+    return sanitizeWithCaja(html);
+  }
+}
+
+/**
+ * @param {string} dirty
+ * @return {string}
+ */
+function purifyHtml(dirty) {
+  const config = Object.assign({}, PURIFY_CONFIG, {
+    'ADD_ATTR': WHITELISTED_ATTRS.concat(PURIFY_CONFIG['ADD_ATTR']),
+    'FORBID_ATTR': isExperimentOn(self, 'inline-styles') ? [] : ['style'],
+    'FORBID_TAGS': Object.keys(BLACKLISTED_TAGS),
+  });
+  DOMPurify.addHook('uponSanitizeElement', uponSanitizeElement);
+  DOMPurify.addHook('uponSanitizeAttribute', uponSanitizeAttribute);
+  return DOMPurify.sanitize(dirty, config);
+}
+
+/**
+ * @param {!Node} node
+ * @param {{tagName: string, allowedTags: !Object<string, boolean>}} data
+ */
+function uponSanitizeElement(node, data) {
+  const {tagName, allowedTags} = data;
+  // Allow all AMP elements (constrained by AMP Validator since tag
+  // calculation is not possible).
+  if (startsWith(tagName, 'amp-')) {
+    allowedTags[tagName] = true;
+  }
+
+  if (tagName == 'a') {
+    if (node.hasAttribute('href') && !node.hasAttribute('target')) {
+      node.setAttribute('target', '_top');
+    }
+  }
+}
+
+/**
+ * @param {!Node} node
+ * @param {{attrName: string, attrValue: string, allowedAttributes: !Array<string>}} data
+ */
+function uponSanitizeAttribute(node, data) {
+  const tagName = (node.tagName || '').toLowerCase();
```

Comment:
I was about to warn here about DOM clobbering, but it turns out DOMPurify just drops clobbered DOM elements whose tagName is unreliable \o/ https://github.com/cure53/DOMPurify/blob/master/src/purify.js#L527 It'll also filter elements with some other clobbered properties. Please check if you only use the properties that are guaranteed to be unclobbered (I've quickly ran through them and didn't see anything you use that is not protected). It's also worth a comment for future updates to this function.